lib/ret_web/channels/hub_channel.ex

```diff
@@ -392,8 +547,68 @@ defmodule RetWeb.HubChannel do
   end
 
   def handle_out("naf" = event, payload, socket) do
+    sender_account = Account |> Ret.Repo.get(payload["sender_account_id"])
+    hub = socket |> hub_for_socket
+
+    {socket, payload, should_broadcast} =
+      case payload do
+        %{"data" => %{"isFirstSync" => true, "creator" => creator, "template" => template, "networkId" => network_id}} ->
+          secure_scene_object = socket.assigns.secure_scene_objects |> Enum.find(&(&1.network_id == network_id))
+          authorized = sender_account |> can?(spawn_and_move_media(hub))
+
+          socket =
+            if authorized and secure_scene_object == nil do
+              socket
+              |> assign(:secure_scene_objects, [
+                %{creator: creator, network_id: network_id, template: template} | socket.assigns.secure_scene_objects
+              ])
+            else
+              socket
+            end
+
+          {payload, sanitized} =
+            if !authorized and secure_scene_object != nil and
+                 secure_scene_object.template |> String.ends_with?("-media") do
+              payload =
+                payload
+                |> Map.put(
+                  "data",
+                  payload["data"] |> sanitize_with_authorized_schemas(secure_scene_object.template, socket)
```

Comment:
this looks like a security hole -- the authorized schemes here being used as far as i can tell are those registered on the receiver's socket, so by sending different authorized schemas on connect the receiver can grant themselves access